### PR TITLE
Add generalised function for checking unit WL status

### PR DESCRIPTION
**Pull request by gnuoy**
Add a more generalised function for checking a units workload status
message. To achieve this two additional functions for resolving
subordinates have been added.

### CODE REVIEW

**Review by AurelienLourot — CHANGES_REQUESTED** [review]
Great, thanks! Just found 2 typos. Happy to land this once they are fixed.

**Review by AurelienLourot — APPROVED** [review]

File: zaza/model.py [hunk 1]
```diff
@@ -1640,6 +1641,91 @@ async def _unit_status():
     async_block_until_wl_status_info_starts_with)
 
 
+async def async_block_until_unit_wl_status_info(
```
Comment (thedac) [inline zaza/model.py:1644]: Can we get a more specific name for the function?
async_block_until_unit_wl_status_regex?

**Review by thedac — CHANGES_REQUESTED** [review]
Code looks good here. But I want to avoid confusion. It is entirely unclear from the function name and the docstring how the new function, async_block_until_unit_wl_status_info is different from async_block_until_unit_wl_status.

Both say a variation of the same thing:
"Block until the unit has a status message that matches pattern."
"Block until the given unit has the desired workload status."

I appreciate you intend to replace the latter function but in the meantime let's make it very clear how they are different. Indicate in the function name and the doctrine that the new function is using regular expressions.

**Review by thedac — APPROVED** [review]
Close enough :+1:

**Pull request merged**